plugins/woocommerce/src/Internal/Admin/Orders/PageController.php

```diff
@@ -67,7 +67,7 @@ private function verify_edit_permission() {
 			wp_die( esc_html__( 'Order type mismatch.', 'woocommerce' ) );
 		}
 
-		if ( ! current_user_can( get_post_type_object( $this->order_type )->cap->edit_post, $this->order->get_id() ) && ! current_user_can( 'manage_woocommerce' ) ) {
+		if ( ! current_user_can( get_post_type_object( $this->order_type )->cap->edit_post, $this->order->get_id() ) && ! current_user_can( 'edit_shop_orders' ) ) {
```

Comment:
Curious if we need that final check/if it is appropriate, since we already check if the current user has the equivalent of `edit_post` (ie, `edit_shop_order`), and meta capability mapping means this should automatically check against either `edit_shop_orders` or `edit_others_shop_orders` as appropriate.

Reply:
Relating [to this comment](https://github.com/woocommerce/woocommerce/pull/50764#issuecomment-2307896319), I think perhaps we both initially misunderstood what's happening here.

As is, we are not requiring the user to have the `manage_woocommerce` capability. Instead, we are saying they have permission if they *do* have that cap ... but they also have permission if the first condition is met.